vdom: Use vdom in topic list.

showell opened the pull request:
This will give us a bit more flexibility for
updating DOM when we have widgets that
get appended to other containers.

Test by opening Private Messages and have another user log on to force an update for the presence circle.

timabbott commented on static/js/pm_list.js:
```diff
@@ -117,7 +117,16 @@ exports.update_private_messages = function () {
         const new_dom = exports._build_private_messages_list();
-        vdom.update(container, new_dom, prior_dom);
+
+        function insert(html) {
```
Why the name `insert`?  I would think something like `replace_content` would be clearer.

showell replied:
fixed

timabbott commented:
I'm test-deploying this on chat.zulip.org.

showell commented:
I updated the PR to fix the bug reported during our first test deployment (active topic not being bolded).  I also fixed a node test coverage issue that was caused by us simplifying topic_list and therefore not exercise some FoldDict code.

showell commented:
Updated again to play nice with dict-related changes that make values() an iterator.

timabbott commented:
This seems close enough to fully reliably that is makes sense to merge this and fix-forward, so merged, thanks @showell!  We're getting as many bugs from rebase issues as actual problems with the implementation at this point.
